### PR TITLE
[CI Fest] Fix flaky performance test `decimal_aggregates`

### LINKED ISSUE #67034: Performance test `decimal_aggregates` is flaky
Sometimes fail with timeout: https://s3.amazonaws.com/clickhouse-test-reports/66761/285cd6c17cc398111d5fd1afeb2e7a3e041fe47d/performance_comparison__release__[4_4]/report.html

```
2024.07.23 09:47:33.099184 [ 1074 ] {092b9169-bbbf-46a1-bceb-261d82c2e4d6} <Debug> executeQuery: (from [::1]:47836) INSERT INTO t SELECT number AS x, x % 1000000 AS d32, x AS d64, x d128 FROM numbers_mt(500000000) SETTINGS max_threads = 8 (stage: Complete)
2024.07.23 09:47:33.099374 [ 1074 ] {092b9169-bbbf-46a1-bceb-261d82c2e4d6} <Trace> Planner: Query to stage Complete
2024.07.23 09:47:33.099457 [ 1074 ] {092b9169-bbbf-46a1-bceb-261d82c2e4d6} <Trace> Planner: Query from stage FetchColumns to stage Complete
2024.07.23 09:47:33.566003 [ 2883 ] {092b9169-bbbf-46a1-bceb-261d82c2e4d6} <Debug> MemoryTracker: Current memory usage (for query): 1.00 GiB.
2024.07.23 09:47:34.017259 [ 2851 ] {092b9169-bbbf-46a1-bceb-261d82c2e4d6} <Debug> MemoryTracker: Current memory usage (for query): 2.00 GiB.
2024.07.23 09:47:34.458223 [ 2855 ] {092b9169-bbbf-46a1-bceb-261d82c2e4d6} <Debug> MemoryTracker: Current memory usage (for query): 3.00 GiB.
2024.07.23 09:47:34.898117 [ 2855 ] {092b9169-bbbf-46a1-bceb-261d82c2e4d6} <Debug> MemoryTracker: Current memory usage (for query): 4.00 GiB.
2024.07.23 09:47:35.338867 [ 2855 ] {092b9169-bbbf-46a1-bceb-261d82c2e4d6} <Debug> MemoryTracker: Current memory usage (for query): 5.00 GiB.
2024.07.23 09:47:35.782748 [ 2678 ] {092b9169-bbbf-46a1-bceb-261d82c2e4d6} <Debug> MemoryTracker: Current memory usage (for query): 6.00 GiB.
2024.07.23 09:47:36.225998 [ 2678 ] {092b9169-bbbf-46a1-bceb-261d82c2e4d6} <Debug> MemoryTracker: Current memory usage (for query): 7.00 GiB.
2024.07.23 09:47:36.667445 [ 2883 ] {092b9169-bbbf-46a1-bceb-261d82c2e4d6} <Debug> MemoryTracker: Current memory usage (for query): 8.00 GiB.
2024.07.23 09:47:37.032047 [ 2904 ] {092b9169-bbbf-46a1-bceb-261d82c2e4d6} <Debug> MemoryTracker: Current memory usage (total): 10.00 GiB.
2024.07.23 09:47:37.109828 [ 2883 ] {092b9169-bbbf-46a1-bceb-261d82c2e4d6} <Debug> MemoryTracker: Current memory usage (for query): 9.00 GiB.
2024.07.23 09:47:37.497420 [ 2868 ] {092b9169-bbbf-46a1-bceb-261d82c2e4d6} <Debug> MemoryTracker: Current memory usage (total): 11.00 GiB.
2024.07.23 09:47:37.581969 [ 2868 ] {092b9169-bbbf-46a1-bceb-261d82c2e4d6} <Debug> MemoryTracker: Current memory usage (for query): 10.00 GiB.
2024.07.23 09:47:37.999828 [ 2851 ] {092b9169-bbbf-46a1-bceb-261d82c2e4d6} <Debug> MemoryTracker: Current memory usage (total): 12.00 GiB.
2024.07.23 09:47:38.083845 [ 2674 ] {092b9169-bbbf-46a1-bceb-261d82c2e4d6} <Debug> MemoryTracker: Current memory usage (for query): 11.00 GiB.
2024.07.23 09:47:38.474548 [ 2678 ] {092b9169-bbbf-46a1-bceb-261d82c2e4d6} <Debug> MemoryTracker: Current memory usage (total): 13.00 GiB.
2024.07.23 09:47:38.554410 [ 2868 ] {092b9169-bbbf-46a1-bceb-261d82c2e4d6} <Debug> MemoryTracker: Current memory usage (for query): 12.00 GiB.
2024.07.23 09:47:38.959971 [ 2674 ] {092b9169-bbbf-46a1-bceb-261d82c2e4d6} <Debug> MemoryTracker: Current memory usage (total): 14.00 GiB.
2024.07.23 09:47:39.043562 [ 2810 ] {092b9169-bbbf-46a1-bceb-261d82c2e4d6} <Debug> MemoryTracker: Current memory usage (for query): 13.00 GiB.
2024.07.23 09:47:39.534853 [ 1074 ] {092b9169-bbbf-46a1-bceb-261d82c2e4d6} <Debug> TCPHandler: Sending profile events block with 21 rows, 1450 bytes took 103 milliseconds
2024.07.23 09:47:43.348860 [ 1074 ] {092b9169-bbbf-46a1-bceb-261d82c2e4d6} <Debug> TCPHandler: Sending profile events block with 18 rows, 1239 bytes took 3515 milliseconds
2024.07.23 09:47:45.317044 [ 1074 ] {092b9169-bbbf-46a1-bceb-261d82c2e4d6} <Debug> TCPHandler: Sending profile events block with 20 rows, 1374 bytes took 1607 milliseconds
2024.07.23 09:47:53.997605 [ 1074 ] {092b9169-bbbf-46a1-bceb-261d82c2e4d6} <Debug> TCPHandler: Sending profile events block with 19 rows, 1311 bytes took 7497 milliseconds
2024.07.23 09:48:02.380465 [ 1074 ] {092b9169-bbbf-46a1-bceb-261d82c2e4d6} <Debug> TCPHandler: Sending profile events block with 9 rows, 626 bytes took 6592 milliseconds
2024.07.23 09:48:12.635689 [ 1074 ] {092b9169-bbbf-46a1-bceb-261d82c2e4d6} <Debug> TCPHandler: Sending profile events block with 17 rows, 1184 bytes took 8399 milliseconds
2024.07.23 09:48:21.657128 [ 1074 ] {092b9169-bbbf-46a1-bceb-261d82c2e4d6} <Debug> TCPHandler: Sending profile events block with 17 rows, 1184 bytes took 7409 milliseconds
2024.07.23 09:48:30.264432 [ 1074 ] {092b9169-bbbf-46a1-bceb-261d82c2e4d6} <Debug> TCPHandler: Sending profile events block with 9 rows, 626 bytes took 6847 milliseconds
2024.07.23 09:48:41.647688 [ 1074 ] {092b9169-bbbf-46a1-bceb-261d82c2e4d6} <Debug> TCPHandler: Sending profile events block with 17 rows, 1184 bytes took 9587 milliseconds
2024.07.23 09:48:50.559362 [ 1074 ] {092b9169-bbbf-46a1-bceb-261d82c2e4d6} <Debug> TCPHandler: Sending profile events block with 9 rows, 626 bytes took 7177 milliseconds
2024.07.23 09:49:00.782216 [ 1074 ] {092b9169-bbbf-46a1-bceb-261d82c2e4d6} <Debug> TCPHandler: Sending profile events block with 17 rows, 1184 bytes took 8359 milliseconds
2024.07.23 09:49:09.531984 [ 1074 ] {092b9169-bbbf-46a1-bceb-261d82c2e4d6} <Debug> TCPHandler: Sending profile events block with 9 rows, 626 bytes took 7002 milliseconds
2024.07.23 09:49:19.545056 [ 1074 ] {092b9169-bbbf-46a1-bceb-261d82c2e4d6} <Debug> TCPHandler: Sending profile events block with 21 rows, 1446 bytes took 8233 milliseconds
2024.07.23 09:49:30.405319 [ 1074 ] {092b9169-bbbf-46a1-bceb-261d82c2e4d6} <Debug> TCPHandler: Sending profile events block with 21 rows, 1446 bytes took 8976 milliseconds
2024.07.23 09:49:40.388765 [ 1074 ] {092b9169-bbbf-46a1-bceb-261d82c2e4d6} <Debug> TCPHandler: Sending profile events block with 20 rows, 1374 bytes took 8254 milliseconds
2024.07.23 09:49:50.568879 [ 1074 ] {092b9169-bbbf-46a1-bceb-261d82c2e4d6} <Debug> TCPHandler: Sending profile events block with 19 rows, 1311 bytes took 8411 milliseconds
2024.07.23 09:50:00.181240 [ 1074 ] {092b9169-bbbf-46a1-bceb-261d82c2e4d6} <Debug> TCPHandler: Sending profile events block with 18 rows, 1240 bytes took 7735 milliseconds
2024.07.23 09:50:12.642769 [ 1074 ] {092b9169-bbbf-46a1-bceb-261d82c2e4d6} <Debug> TCPHandler: Sending profile events block with 19 rows, 1311 bytes took 10488 milliseconds
2024.07.23 09:50:24.001033 [ 1074 ] {092b9169-bbbf-46a1-bceb-261d82c2e4d6} <Debug> TCPHandler: Sending profile events block with 18 rows, 1240 bytes took 9529 milliseconds
2024.07.23 09:50:35.967332 [ 1074 ] {092b9169-bbbf-46a1-bceb-261d82c2e4d6} <Debug> TCPHandler: Sending profile events block with 20 rows, 1374 bytes took 10212 milliseconds
2024.07.23 09:50:45.171842 [ 1074 ] {092b9169-bbbf-46a1-bceb-261d82c2e4d6} <Debug> TCPHandler: Sending profile events block with 19 rows, 1311 bytes took 7606 milliseconds
2024.07.23 09:50:56.333619 [ 1074 ] {092b9169-bbbf-46a1-bceb-261d82c2e4d6} <Debug> TCPHandler: Sending profile events block with 20 rows, 1374 bytes took 9652 milliseconds
2024.07.23 09:51:06.476411 [ 1074 ] {092b9169-bbbf-46a1-bceb-261d82c2e4d6} <Debug> TCPHandler: Sending profile events block with 19 rows, 1311 bytes took 8430 milliseconds
2024.07.23 09:51:16.596482 [ 1074 ] {092b9169-bbbf-46a1-bceb-261d82c2e4d6} <Debug> TCPHandler: Sending profile events block with 19 rows, 1311 bytes took 8485 milliseconds
2024.07.23 09:51:25.825948 [ 1074 ] {092b9169-bbbf-46a1-bceb-261d82c2e4d6} <Debug> TCPHandler: Sending profile events block with 19 rows, 1311 bytes took 7711 milliseconds
2024.07.23 09:51:36.223628 [ 1074 ] {092b9169-bbbf-46a1-bceb-261d82c2e4d6} <Debug> TCPHandler: Sending profile events block with 19 rows, 1311 bytes took 8931 milliseconds
2024.07.23 09:51:45.938104 [ 1074 ] {092b9169-bbbf-46a1-bceb-261d82c2e4d6} <Debug> TCPHandler: Sending profile events block with 16 rows, 1113 bytes took 8142 milliseconds
2024.07.23 09:51:57.218862 [ 1074 ] {092b9169-bbbf-46a1-bceb-261d82c2e4d6} <Debug> TCPHandler: Sending profile events block with 17 rows, 1184 bytes took 9261 milliseconds
2024.07.23 09:52:07.762086 [ 1074 ] {092b9169-bbbf-46a1-bceb-261d82c2e4d6} <Debug> TCPHandler: Sending profile events block with 16 rows, 1113 bytes took 8852 milliseconds
2024.07.23 09:52:19.236125 [ 1074 ] {092b9169-bbbf-46a1-bceb-261d82c2e4d6} <Debug> TCPHandler: Sending profile events block with 19 rows, 1311 bytes took 9467 milliseconds
2024.07.23 09:52:29.361438 [ 1074 ] {092b9169-bbbf-46a1-bceb-261d82c2e4d6} <Debug> TCPHandler: Sending profile events block with 17 rows, 1184 bytes took 8439 milliseconds
2024.07.23 09:52:40.664238 [ 1074 ] {092b9169-bbbf-46a1-bceb-261d82c2e4d6} <Debug> TCPHandler: Sending profile events block with 17 rows, 1184 bytes took 9319 milliseconds
2024.07.23 09:52:49.109412 [ 1074 ] {092b9169-bbbf-46a1-bceb-261d82c2e4d6} <Debug> TCPHandler: Sending profile events block with 8 rows, 563 bytes took 6855 milliseconds
2024.07.23 09:52:58.299029 [ 1074 ] {092b9169-bbbf-46a1-bceb-261d82c2e4d6} <Debug> TCPHandler: Sending profile events block with 6 rows, 433 bytes took 7543 milliseconds
2024.07.23 09:53:06.559293 [ 1074 ] {092b9169-bbbf-46a1-bceb-261d82c2e4d6} <Debug> TCPHandler: Sending profile events block with 6 rows, 433 bytes took 6553 milliseconds
2024.07.23 09:53:16.751997 [ 1074 ] {092b9169-bbbf-46a1-bceb-261d82c2e4d6} <Debug> TCPHandler: Sending profile events block with 14 rows, 991 bytes took 8540 milliseconds
2024.07.23 09:53:25.086920 [ 1074 ] {092b9169-bbbf-46a1-bceb-261d82c2e4d6} <Debug> TCPHandler: Sending profile events block with 13 rows, 920 bytes took 6890 milliseconds
2024.07.23 09:53:33.293999 [ 1074 ] {092b9169-bbbf-46a1-bceb-261d82c2e4d6} <Debug> TCPHandler: Sending profile events block with 6 rows, 433 bytes took 6616 milliseconds
2024.07.23 09:53:41.599625 [ 1074 ] {092b9169-bbbf-46a1-bceb-261d82c2e4d6} <Debug> TCPHandler: Sending profile events block with 13 rows, 920 bytes took 6697 milliseconds
2024.07.23 09:53:49.382091 [ 1074 ] {092b9169-bbbf-46a1-bceb-261d82c2e4d6} <Debug> TCPHandler: Sending profile events block with 15 rows, 1054 bytes took 6302 milliseconds
2024.07.23 09:53:56.066377 [ 1074 ] {092b9169-bbbf-46a1-bceb-261d82c2e4d6} <Debug> TCPHandler: Sending profile events block with 6 rows, 433 bytes took 5190 milliseconds
2024.07.23 09:54:01.275142 [ 1074 ] {092b9169-bbbf-46a1-bceb-261d82c2e4d6} <Debug> TCPHandler: Sending profile events block with 6 rows, 433 bytes took 4465 milliseconds
2024.07.23 09:54:02.511588 [ 1074 ] {092b9169-bbbf-46a1-bceb-261d82c2e4d6} <Debug> TCPHandler: Sending profile events block with 17 rows, 1189 bytes took 875 milliseconds
2024.07.23 09:54:06.239564 [ 1074 ] {092b9169-bbbf-46a1-bceb-261d82c2e4d6} <Debug> TCPHandler: Sending profile events block with 6 rows, 433 bytes took 3363 milliseconds
2024.07.23 09:54:12.782668 [ 1074 ] {092b9169-bbbf-46a1-bceb-261d82c2e4d6} <Debug> TCPHandler: Sending profile events block with 6 rows, 433 bytes took 5120 milliseconds
2024.07.23 09:54:14.092850 [ 1074 ] {092b9169-bbbf-46a1-bceb-261d82c2e4d6} <Debug> TCPHandler: Sending profile events block with 6 rows, 433 bytes took 352 milliseconds
2024.07.23 09:54:16.619833 [ 1074 ] {092b9169-bbbf-46a1-bceb-261d82c2e4d6} <Debug> TCPHandler: Sending profile events block with 6 rows, 433 bytes took 1762 milliseconds
2024.07.23 09:54:19.986332 [ 1074 ] {092b9169-bbbf-46a1-bceb-261d82c2e4d6} <Debug> TCPHandler: Sending profile events block with 6 rows, 433 bytes took 2139 milliseconds
2024.07.23 09:54:21.767033 [ 1074 ] {092b9169-bbbf-46a1-bceb-261d82c2e4d6} <Debug> TCPHandler: Sending profile events block with 17 rows, 1182 bytes took 960 milliseconds
2024.07.23 09:54:22.374481 [ 1074 ] {092b9169-bbbf-46a1-bceb-261d82c2e4d6} <Debug> TCPHandler: Sending profile events block with 6 rows, 433 bytes took 244 milliseconds
2024.07.23 09:54:23.264940 [ 1074 ] {092b9169-bbbf-46a1-bceb-261d82c2e4d6} <Debug> TCPHandler: Sending profile events block with 6 rows, 433 bytes took 465 milliseconds
2024.07.23 09:54:24.990702 [ 1074 ] {092b9169-bbbf-46a1-bceb-261d82c2e4d6} <Debug> TCPHandler: Sending profile events block with 6 rows, 433 bytes took 1476 milliseconds
2024.07.23 09:54:26.102469 [ 1074 ] {092b9169-bbbf-46a1-bceb-261d82c2e4d6} <Debug> TCPHandler: Sending profile events block with 6 rows, 433 bytes took 689 milliseconds
2024.07.23 09:54:27.044493 [ 1074 ] {092b9169-bbbf-46a1-bceb-261d82c2e4d6} <Debug> TCPHandler: Sending profile events block with 6 rows, 433 bytes took 583 milliseconds
2024.07.23 09:54:27.952582 [ 1074 ] {092b9169-bbbf-46a1-bceb-261d82c2e4d6} <Debug> TCPHandler: Sending profile events block with 6 rows, 433 bytes took 574 milliseconds
2024.07.23 09:54:35.206300 [ 1074 ] {092b9169-bbbf-46a1-bceb-261d82c2e4d6} <Error> executeQuery: Code: 159. DB::Exception: Timeout exceeded: elapsed 405.944779779 seconds, maximum: 300. (TIMEOUT_EXCEEDED) (version 24.7.1.2843 (official build)) (from [::1]:47836) (in query: INSERT INTO t SELECT number AS x, x % 1000000 AS d32, x AS d64, x d128 FROM numbers_mt(500000000) SETTINGS max_threads = 8), Stack trace (when copying this message, always include the lines below):
2024.07.23 09:54:35.249883 [ 1074 ] {092b9169-bbbf-46a1-bceb-261d82c2e4d6} <Error> TCPHandler: Code: 159. DB::Exception: Timeout exceeded: elapsed 405.944779779 seconds, maximum: 300. (TIMEOUT_EXCEEDED), Stack trace (when copying this message, always include the lines below):
2024.07.23 09:54:35.250614 [ 1074 ] {092b9169-bbbf-46a1-bceb-261d82c2e4d6} <Debug> TCPHandler: Processed in 422.1457601 sec.
2024.07.23 09:54:35.250659 [ 1074 ] {092b9169-bbbf-46a1-bceb-261d82c2e4d6} <Debug> MemoryTracker: Peak memory usage (for query): 13.76 GiB.
```

### PR DESCRIPTION
**Pull request by alesapin**
### Changelog category (leave one):
- Not for changelog (changelog entry is not required)

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Split query into multiple queries to consume less memory at once + use less data. Fixes #67034.

### Documentation entry for user-facing changes

- [ ] Documentation is written (mandatory for new features)

> Information about CI checks: https://clickhouse.com/docs/en/development/continuous-integration/

#### CI Settings (Only check the boxes if you know what you are doing):
- [ ]  Allow: All Required Checks
- [ ]  Allow: Stateless tests
- [ ]  Allow: Stateful tests
- [ ]  Allow: Integration Tests
- [x]  Allow: Performance tests
- [ ]  Allow: All Builds
- [ ]  Allow: batch 1, 2 for multi-batch jobs
- [ ]  Allow: batch 3, 4, 5, 6 for multi-batch jobs
---
- [ ]  Exclude: Style check
- [ ]  Exclude: Fast test
- [ ]  Exclude: All with ASAN
- [ ]  Exclude: All with TSAN, MSAN, UBSAN, Coverage
- [ ]  Exclude: All with aarch64, release, debug
---
- [ ]  Do not test
- [ ]  Woolen Wolfdog
- [ ]  Upload binaries for special builds
- [ ]  Disable merge-commit
- [ ]  Disable CI cache

### CODE REVIEW

**Comment by robot-ch-test-poll4** [comment]
*This is an automated comment for commit a03a59a3bfd2dc55f1f04197cad36d147edb3ee1 with description of existing statuses. It's updated for the latest CI running*

[✅ Click here](https://s3.amazonaws.com/clickhouse-test-reports/67044/a03a59a3bfd2dc55f1f04197cad36d147edb3ee1/ci_running.html) to open a full report in a separate page

<details><summary>Successful checks</summary>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>Docs check</td><td>Builds and tests the documentation</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/65521/9d7bb2a430b142e0761cc30efcf78ea1bbeb6871/docs_check.html">✅ success</a></td></tr>
<tbody>
</table>
</details>

**Review by vdimir — APPROVED** [review]

**Comment by alesapin** [comment]
Aaand not it's unstable 
image

File: tests/performance/decimal_aggregates.xml [hunk 1]
```diff
@@ -4,8 +4,11 @@
     </settings>
 
     <create_query>CREATE TABLE t (x UInt64, d32 Decimal32(3), d64 Decimal64(4), d128 Decimal128(5)) ENGINE = Memory</create_query>
-    <!-- use less threads to save memory -->
-    <fill_query>INSERT INTO t SELECT number AS x, x % 1000000 AS d32, x AS d64, x d128 FROM numbers_mt(500000000) SETTINGS max_threads = 8</fill_query>
+    <!-- use less threads and several queries to save memory -->
+    <fill_query>INSERT INTO t SELECT number AS x, x % 1000000 AS d32, x AS d64, x d128 FROM numbers_mt(100000000) SETTINGS max_threads = 2</fill_query>
+    <fill_query>INSERT INTO t SELECT number AS x, x % 1000000 AS d32, x AS d64, x d128 FROM numbers_mt(100000000, 100000000) SETTINGS max_threads = 2</fill_query>
+    <fill_query>INSERT INTO t SELECT number AS x, x % 1000000 AS d32, x AS d64, x d128 FROM numbers_mt(200000000, 100000000) SETTINGS max_threads = 2</fill_query>
```
Comment (alesapin) [inline tests/performance/decimal_aggregates.xml:10]:
```suggestion
    <fill_query>INSERT INTO t SELECT number AS x, x % 1000000 AS d32, x AS d64, x d128 FROM numbers_mt(200000000, 100000000) SETTINGS max_threads = 2</fill_query>
     <fill_query>INSERT INTO t SELECT number AS x, x % 1000000 AS d32, x AS d64, x d128 FROM numbers_mt(300000000, 100000000) SETTINGS max_threads = 2</fill_query>
```

File: tests/performance/decimal_aggregates.xml [hunk 2]
```diff
@@ -4,8 +4,12 @@
...
+    <fill_query>INSERT INTO t SELECT number AS x, x % 1000000 AS d32, x AS d64, x d128 FROM numbers_mt(300000000, 100000000) SETTINGS max_threads = 2</fill_query>
```
Comment (alesapin) [inline tests/performance/decimal_aggregates.xml:11]:
```suggestion
    <fill_query>INSERT INTO t SELECT number AS x, x % 1000000 AS d32, x AS d64, x d128 FROM numbers_mt(300000000, 100000000) SETTINGS max_threads = 2</fill_query>
    <fill_query>INSERT INTO t SELECT number AS x, x % 1000000 AS d32, x AS d64, x d128 FROM numbers_mt(400000000, 100000000) SETTINGS max_threads = 2</fill_query>
```

**Comment by alesapin** [comment]
Okay, two unstable but at least it doesn't crash with OOM

**Pull request merged**